packages/aws_common/lib/src/http/aws_http_client.dart

```diff
@@ -131,8 +131,12 @@ abstract class AWSBaseHttpClient extends AWSCustomHttpClient {
     responseProgressCompleter.setSourceStream(operation.responseProgress);
     operation.operation.then(
       (resp) async {
-        resp = await transformResponse(resp);
-        completer.complete(resp);
+        try {
+          resp = await transformResponse(resp);
+          completer.complete(resp);
+        } on Exception catch (e, st) {
```

Comment:
Yeah that's fine, but let's add some tests since I didn't understand that about `CancelableOperation.then`

Also, let's handle all errors and exceptions this way so that errors don't get lost either.

```suggestion
        } on Object catch (e, st) {
```

Reply:
Made this change, plus added some unit tests in aws_common to test this behavior in request/response transform (though the REST tests in API category would fail as well without this change).